### ISSUE TITLE
Value of PEG at launch, and tPEG

### ISSUE THREAD

**Issue by WhoSoup** [pegnet/pegnet#124]
Didn't see an issue for this and this discussion comes up frequently on discord: What exactly is the value of PNT in the OPR going to be?

In the proposal paper, the value of PNT is described as `sum(PNT) = sum(all assets in usd) / 4`, but recently @PaulSnow has indicated that it might be better to not give PNT a floating value and to wait for PNT to be listed by exchanges so we can mine those exchanges like everything else:

![image](https://user-images.githubusercontent.com/34172041/61783444-19f57980-ae08-11e9-9e09-5d82fff2af85.png)

I'm not sure if this should be a milestone 1 or 2 problem, but it ties in with https://github.com/pegnet/pegnet/issues/11

**Comment by PaulSnow** [comment]
Common.go defines a burn address for the PegNet MainNet and the PegNet TestNet.

"main": "EC1moooFCT2mainoooo1oooo1oooo1oooo1oooo1oooo1opfDJqF",
"test": "EC1moooFCT2TESToooo1oooo1oooo1oooo1oooo1oooo1on1iNDk",

**Comment by WhoSoup** [comment]
Looks like we'll need to implement the following for milestone1:
* Append the monitor to also watch for transactions to the addresses listed in Paul's comment above
* Calculate how much FCT they overpaid in the fee
* Accumulate those overpays as balances counting toward that FA address
* Have the miners calculate the value of PNT as `sum(all PNT) = sum(pFCT)` 
* Calculate the value of 1 PNT in regards to pUSD and make it available to the mining threads so they can use that value in the OPR

Optional:
* write the code in such a way that it can be reused for milestone2/3 when we add conversions and transactions

**Comment by Cahl-Dee** [comment]
On the call, we did not have anyone representing the opinion that PNT needed a value for pM1 launch. We're going to do all work up to that but do not plan to do the whole thing until there is an argument made to do so.

This means we will do:
- Append the monitor to also watch for transactions to the addresses listed in Paul's comment above
- Calculate how much FCT they overpaid in the fee
- Accumulate those overpays as balances counting toward that FA address

And will put off:
- Have the miners calculate the value of PNT as sum(all PNT) = sum(pFCT)
- Calculate the value of 1 PNT in regards to pUSD and make it available to the mining threads so they can use that value in the OPR
- write the code in such a way that it can be reused for milestone2/3 when we add conversions and transactions (and by reused i mean it will likely be done in FAT)

**Comment by Emyrk** [comment]
I believe we should keep the value of PNT at 0 for at least the first phase. By keeping it at 0, we are telling miners and people onlooking that the price is still speculative. 

If we calculate the price immediately, and we don't see a lot of initial burning, the price of PNT will be some value close to 0 as miners inflate PNT. This is telling the miners their PNT is worth effectively nothing, and in a pegged network, that feels odd.

I just think we need to bootstrap the network right, and we should make the calculations at that point.

**Comment by PaulSnow** [comment]
PNT is actually the name of the token on the MainNet by most of the documentation (mostly assumed), and the TestNet should use the same method to value tPNT as the PNT is valued at launch. until exchanges establish a value for PNT.  Note that tPNT will never be on exchanges, so it will use the method we set up forever.

**Comment by PaulSnow** [comment]
If the PNT isn't actually valued until we turn on conversions later, then we don't have to do anything here, and this issue can be moved to Milestone 2.

**Comment by DavidChapman77** [comment]
We should report.  There is an equation to determine PNTs value.  Reporting as 0 when that equation exists is not correct.  Every token has some speculative value prior to it hitting an exchange where the market determines its value.

**Comment by PaulSnow** [comment]
Well, I don't disagree with David.  I do think that everyone will be able to figure out the pegged value of the PNT anyway (as we document it), and we will have to report the value to do conversions, which are quickly following milestone 1.

Right now the little poll I put on discord is trending towards not reporting.  As it takes a little bit off the plates of the developers, I am not opposed waiting.

**Comment by Cahl-Dee** [comment]
Moving to pM2 as per discussions and vote in Discord.
![image](https://user-images.githubusercontent.com/1673298/62064364-adf58580-b1fa-11e9-9dd9-1382e8543aa8.png)

**Comment by Cahl-Dee** [comment]
Until we have an exchange value we can query and price the PEG, PEG will be priced at the value of the assets / supply of the PEG.

When the time comes for us to start pulling in the data from exchanges, we will want to consider a minimum volume. 

Closing this as we've reached the above decision.

**Comment by Cahl-Dee** [comment]
@PaulSnow @Emyrk 

1. Is there a separate issue for the code to initially value PEG with the pFCT market cap / supply?
2. Have we determined how that'll then transition to exchange-based valuation and how that transition will happen / criteria?

Point 2 can probably wait till pM3, but point 1 is needed for pM2 if I'm not mistaken.

**Comment by PaulBernier** [comment]
**Suggestion: do not artifically set a price for PEG until listed**

Myself and others are worried about setting the price of PEG via a formula. First the exact formula has not been officially published (which data price is taken from which block? with supply taken from which block? current? previous?) which is annoying 11 days before the planned release. Second there are many unknown market dynamics that will be at play with little benefits in return in our opinion. Keeping the price set to $0 until being listed doesn't prevent the system from working normally and being tested as all owners of pFCT can enjoy the PegNet, they just cannot convert to PEG (which is ok as there is no market for PEG anyway). It also adds an extra incentive to get pFCT as the only way to enjoy the PegNet utility.

In our view, here are the pros and cons of setting PEG price via a formula:

Pros:

    Allow miners to acquire pAssets immediately.

Cons:

    formula will set the initial price of PEG at around $0.003 ($100,000 / 30M PEG) which is a very low price making mining for most people far from profitable. It sends a bad signal and potentially anchor a low price in people mind (which could be reflected on the future exchange price).
    we will have 2 completely different market dynamics in a short succession, some actors may profit a lot or lose a lot in one of those market dynamics and not the other, making people bitter.
    the transition between formula-based price and market price is unknown, as far as I am aware nobody has taken the time to think about the economics consequence of that transition, especially during the period of time PEG will have 2 values, one by formula and one by market pricing and that one may influence the other. And ultimately when the switch happen, a sudden change of the price could again shake the whole system.
    with this approach the miners need to access the PEG supply and market caps, thus having to call fatd, increasing yet again the complexity of a set up already steep.

In a nutshell, setting the price of PEG early doesn't bring too much real benefit while opening multiple risky unknowns.

**Comment by WhoSoup** [comment]
> In a nutshell, setting the price of PEG early doesn't bring much real benefit while opening multiple risky unknowns.

I think the benefit is a really big one from a marketing/social point of view. Much of the appeal and selling point of PegNet is the ability to convert between pegged assets. That will not be possible for miners until PEG has a value, so when we launch Conversions the only people who are able to enjoy that aspect of PegNet are people who have burnt FCT. 

That may be a big blow to all miners who feel left out. I do not know if it outweighs the economic and technical downsides but it would make sense to include the miner's perspective in this decision.

**Comment by PaulBernier** [comment]
I should have made it clear that I was speaking here as a miner myself and presenting the concerns of a group of miner :) Provided that we are the only one who expressed those concerns and that we are just 1 week away from the launch, I guess we will move forward with the initial plan.

**Comment by PaulSnow** [comment]
At this point, we are setting the value of the PEG = <capitalization of the pegged assets> / <supply of PEG>

So in more detail:

p(n, PEG) = [ Σ[a∈A] p(n+1,a)*s(n-1,a) ] / s(n-1,PEG)

p(x,y) -- Price at block height x of asset y
s(x,y) -- Supply at block height x of asset y
A -- the set of all pegged assets excluding PEG
Σ[a∈A] -- Summation over all a members of A

Supplies are set in the previous block (because changes to the supply by conversions cannot be used to impact price without risk of gaming the process) and prices are set by the next block (because conversions cannot be based on past prices without risk of gaming the process)
